cmd/integrationArtifactUpload_test.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type integrationArtifactUploadMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newIntegrationArtifactUploadTestsUtils() integrationArtifactUploadMockUtils {
+	utils := integrationArtifactUploadMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunIntegrationArtifactUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Create Test", func(t *testing.T) {
+
+		tempDir, tmpErr := ioutil.TempDir("", "")
+		defer os.RemoveAll(tempDir) // clean up
+		assert.NoError(t, tmpErr, "Error when creating temp dir")
+		files := &mock.FilesMock{}
+		path := filepath.Join(tempDir, "iflow4.zip")
+		files.AddFile(path, []byte("dummy content"))
+		createErr := ioutil.WriteFile(path, []byte("dummy content"), 0755)
```

Comment:
... and here we write file file into the real file system. --> we should use the file system mock only.